packages/typespec-vscode/src/extension.ts

```diff
@@ -1,24 +1,18 @@
-import { ResolveModuleHost } from "@typespec/compiler/module-resolver";
-import { readFile, realpath, stat } from "fs/promises";
-import { dirname, isAbsolute, join, resolve } from "path";
-import vscode, { ExtensionContext, commands, workspace } from "vscode";
-import {
-  Executable,
-  ExecutableOptions,
-  LanguageClient,
-  LanguageClientOptions,
-} from "vscode-languageclient/node.js";
-import logger from "./extension-logger.js";
-import { TypeSpecLogOutputChannel } from "./typespec-log-output-channel.js";
-import { normalizeSlash, useShellInExec } from "./utils.js";
-
-let client: LanguageClient | undefined;
+import vscode, { commands, ExtensionContext } from "vscode";
```

Comment:
refactor to move code to corresponding separated files to avoid one big file contains everything especially when we are going to add more features in vscode according to the plan. 
The code to handle config change is also added, please check the comment below.